src/outmdsd/TraceMacros.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+#ifndef __TRACEMACROS_H__
+#define __TRACEMACROS_H__
+
+#define ADD_INFO_TRACE \
+    Trace _trace(TraceLevel::Info, __func__, __FILE__, __LINE__)
+/**/
+#define ADD_DEBUG_TRACE \
+    Trace _trace(TraceLevel::Debug, __func__, __FILE__, __LINE__)
+/**/
+#define ADD_TRACE_TRACE \
+    Trace _trace(TraceLevel::Trace, __func__, __FILE__, __LINE__)
+/**/
+#define Log(level, message) \
+    if (level >= Trace::GetTraceLevel()) { \
+        std::ostringstream _ss; \
+        _ss << message; \
+        Trace::WriteLog(level, _ss.str(), __FILE__, __LINE__); \
+    } \
+/**/
```

Comment:
I agree that 'Log' is not a good macro name. With carefully designed names, it has less chance of conflicts but won't eliminate name conflicts. Because these macros are intended for 'outmdsd' internal usage only, a safer way is to isolate them here. so that applications using this SDK don't have to include this header when they try to set the Tracer .

---
In reply to: [97008338](https://github.com/Azure/fluentd-plugin-mdsd/pull/2#discussion_r97008338) [](ancestors = 97008338)